### ISSUE TITLE
Allow agent downgrade via Fleet UI

### ISSUE THREAD

**Issue by jvalente-salemstate** [elastic/beats#37619]
**Describe the enhancement:** In addition to upgrading agents when an update is available, allow for downgrade, within reasonable technical restrictions, to an earlier version of Elastic Agent.

**Describe a specific use case for the enhancement or feature:** The metricbeat memory leak is one example. The workarounds pending that fix were to either disable affected modules (and the scope was unclear enough that I disabled System Metrics but it took another week or two to see I needed to also do it in the Agent integration) or manually downgrading. While some customers may have the capacity to do this with a GPO, Ansible, etc, for others this may be doing this manually or other less ideal methods that sort of defeat one point of centrally managing once the initial deployment is done. A downgrade option would be great for this or other bugs/breaking changes.

Another use case, but maybe not applicable depending on what the root cause was, is the instance or few where a specific version fails to upgrade:

https://github.com/elastic/fleet-server/issues/2383

Perhaps this also means downgrades wouldn't work but if not, folks who've upgraded can then downgrade until a fixed version is released without the other workarounds.

**Comment by jvalente-salemstate** [comment]
I see I somehow posted this in the beats repo and not fleet-server where I intended. If that's an issue, this can be closed and I'll re-file there.

**Comment by jlind23** [comment]
Thanks @jvalente-salemstate for creating this. Anything that is Fleet UI related has to be created in the Kibana repository, however I feel like this issue is probably a duplicate of https://github.com/elastic/kibana/issues/172745.
If we think that's right, let's close this one then.

**Comment by jvalente-salemstate** [comment]
@jlind23 correct. It seems to be a duplicate. I'd checked the fleet repo but not the Kibana one. This can be closed as that exists.
